app/services/story_updater_service.rb

```diff
@@ -13,6 +16,11 @@ def save
       assign_attributes
       story.save!
       story.changesets.create!
+
+      fix_project_start_date
+      fix_story_accepted_at
+
+      notify_state_changed
```

Comment:
Even though this is indeed better than the old observers approach, i still think this notification logic could have it's own class that could be called from here instead.